File: scripts/render-readmitted-charts.py
#!/usr/bin/env python3
"""Export readmitted charts as PNG + Plotly JSON (FRBSF-style academic palette)."""

from __future__ import annotations

from pathlib import Path

import pandas as pd
import plotly.graph_objects as go
import plotly.io as pio

ROOT = Path(__file__).resolve().parents[1]
REPO = ROOT / ".cache/article-repos/readmitted"
SITE_DATA = ROOT / "public/data/articles/readmitted/charts"
SITE_IMG = ROOT / "public/images/content/articles/readmitted/charts"

# FRBSF-style academic palette
FED_PRIMARY = "#1D4E77"
FED_OLIVE = "#8B9A46"
FED_RED = "#A52A2A"
FED_CYAN = "#7BC0D3"
FED_GREY_TEXT = "#666666"
FED_GRID = "#E0E0E0"
FED_BG = "#FFFFFF"
FED_MUTED_BLUE = "#B8C9D9"
FED_DARK = "#333333"

FONT = "Helvetica, Arial, sans-serif"

BAR_GRADIENT = [
    "#B8C9D9",
    "#8FAFC4",
    "#6B95AF",
    "#4D7B9A",
    "#356283",
    "#2A5575",
    "#1D4E77",
    "#1A4569",
    "#173C5B",
    "#14334D",
]

TIER_COLORS = {
    "No Penalty": FED_MUTED_BLUE,
    "Low": FED_CYAN,
    "Medium": FED_OLIVE,
    "High": FED_RED,
}

NAT_AVG_PCT = 48.1
HRRP_SOURCE = (
    "CMS Hospital Readmissions Reduction Program (HRRP), "
    "FY2025 supplemental data (Dataset 9n3s-kdb3)"
)


def fed_figure_title(figure_num: int, title: str) -> str:
    return (
        f"<span style='color:{FED_GREY_TEXT};font-size:11px;font-weight:normal;'>"
        f"Figure {figure_num}</span><br>"
        f"<span style='color:{FED_PRIMARY};font-size:16px;font-weight:bold;'>{title}</span>"
    )


def fed_caption(note: str, source: str) -> str:
    return f"<span style='color:{FED_GREY_TEXT};font-size:10px;'>Note: {note}<br>Source: {source}</span>"


def base_layout(
    figure_num: int,
    title: str,
    note: str,
    source: str = HRRP_SOURCE,
    **kwargs,
) -> dict:
    layout = dict(
        title=dict(
            text=fed_figure_title(figure_num, title),
            x=0,
            xanchor="left",
            y=0.98,
            yanchor="top",
        ),
        paper_bgcolor=FED_BG,
        plot_bgcolor=FED_BG,
        font=dict(color=FED_DARK, family=FONT, size=12),
        margin=dict(l=88, r=56, t=96, b=110),
        annotations=[],
    )

    extra_annotations = kwargs.pop("annotations", [])
    layout["annotations"] = [
        dict(
            x=0.01,
            y=0.03,
            xref="paper",
            yref="paper",
            text=fed_caption(note, source),
            showarrow=False,
            xanchor="left",
            yanchor="bottom",
            align="left",
        ),
        dict(
            x=0.99,
            y=0.03,
            xref="paper",
            yref="paper",
            text="ARTOMETRICS",
            showarrow=False,
            xanchor="right",
            yanchor="bottom",
            font=dict(family=FONT, size=12, color="#1C1C1E"),
        ),
        *extra_annotations,
    ]

    extra_shapes = kwargs.pop("shapes", [])
    layout["shapes"] = extra_shapes
    layout.update(kwargs)
    return layout


def save_chart(fig: go.Figure, name: str) -> None:
    for dest in [REPO / "charts", SITE_DATA]:
        dest.mkdir(parents=True, exist_ok=True)
        pio.write_json(fig, dest / f"{name}.plotly.json")

    for dest in [REPO / "charts", SITE_IMG]:
        dest.mkdir(parents=True, exist_ok=True)
        fig.write_image(dest / f"{name}.png", width=1200, height=700, scale=2)
        print(f"Wrote {name} assets")


def chart1(state_df: pd.DataFrame) -> go.Figure:
    df = state_df[state_df["above_nat_avg"]].sort_values("pct_penalized")
    n = len(df)
    if n <= len(BAR_GRADIENT):
        colors = BAR_GRADIENT[:n]
    else:
        colors = [
            BAR_GRADIENT[int(i * (len(BAR_GRADIENT) - 1) / max(n - 1, 1))]
            for i in range(n)
        ]
    colors[-1] = FED_RED

    fig = go.Figure(
        go.Bar(
            x=df["pct_penalized"],
            y=df["state"],
            orientation="h",
            marker=dict(color=colors, line=dict(width=0)),
            text=[f"{v:.1f}%" for v in df["pct_penalized"]],
            textposition="outside",
            textfont=dict(family=FONT, size=10, color=FED_DARK),
            cliponaxis=False,
            hovertemplate="<b>%{y}</b><br>%{x:.1f}% of pairs penalized<extra></extra>",
        )
    )

    annotations = [
        dict(
            x=NAT_AVG_PCT,
            y=1.02,
            xref="x",
            yref="paper",
            text=f"National average ({NAT_AVG_PCT}%)",
            showarrow=False,
            xanchor="center",
            yanchor="bottom",
            font=dict(family=FONT, size=9, color=FED_OLIVE),
        ),
        dict(
            x=df["pct_penalized"].iloc[-1],
            y=df["state"].iloc[-1],
            xref="x",
            yref="y",
            text="NJ leads at 65.4%",
            showarrow=True,
            arrowhead=2,
            arrowsize=0.8,
            arrowwidth=1,
            arrowcolor=FED_RED,
            ax=40,
            ay=-28,
            font=dict(family=FONT, size=9, color=FED_RED),
        ),
    ]

    fig.update_layout(
        **base_layout(
            1,
            "States with above-average HRRP penalty exposure",
            (
                "Panel shows states where the share of hospital-condition pairs with an "
                "excess readmission ratio above 1.0 exceeds the 48.1% national average."
            ),
            xaxis=dict(
                title="Share of hospital-condition pairs with ERR > 1.0 (%)",
                range=[0, max(df["pct_penalized"]) * 1.14],
                gridcolor=FED_GRID,
                gridwidth=0.5,
                zeroline=False,
                showline=True,
                linecolor=FED_DARK,
                linewidth=1,
            ),
            yaxis=dict(
                title="",
                categoryorder="array",
                categoryarray=list(df["state"]),
                showgrid=False,
            ),
            shapes=[
                dict(
                    type="line",
                    x0=NAT_AVG_PCT,
                    x1=NAT_AVG_PCT,
                    y0=0,
                    y1=1,
                    yref="paper",
                    line=dict(color=FED_OLIVE, dash="dash", width=1.5),
                )
            ],
            annotations=annotations,
            height=520 + n * 22 + 80,
        )
    )
    return fig


def chart2() -> go.Figure:
    df = pd.DataFrame(
        {
            "condition": ["Hip/Knee", "COPD", "Heart Failure", "AMI", "Pneumonia", "CABG"],
            "avg_err": [1.00485, 1.00271, 1.00254, 1.00212, 1.00198, 1.00187],
        }
    )
    df = df.sort_values("avg_err")

    fig = go.Figure()
    for _, row in df.iterrows():
        fig.add_trace(
            go.Scatter(
                x=[1.0, row["avg_err"]],
                y=[row["condition"], row["condition"]],
                mode="lines",
                line=dict(color=FED_GRID, width=2),
                showlegend=False,
                hoverinfo="skip",
            )
        )

    colors = [FED_RED if c == "Hip/Knee" else FED_PRIMARY for c in df["condition"]]
    fig.add_trace(
        go.Scatter(
            x=df["avg_err"],
            y=df["condition"],
            mode="markers+text",
            marker=dict(color=colors, size=11, line=dict(width=0.5, color=FED_DARK)),
            text=[f"{v:.5f}" for v in df["avg_err"]],
            textposition="middle right",
            textfont=dict(family=FONT, size=10, color=FED_DARK),
            hovertemplate="<b>%{y}</b><br>ERR: %{x:.5f}<extra></extra>",
            showlegend=False,
        )
    )

    annotations = [
        dict(
            x=1.00485,
            y="Hip/Knee",
            xref="x",
            yref="y",
            text="Nearly 2× the excess<br>of the next condition",
            showarrow=True,
            arrowhead=2,
            arrowsize=0.8,
            arrowwidth=1,
            arrowcolor=FED_RED,
            ax=55,
            ay=-35,
            font=dict(family=FONT, size=9, color=FED_RED),
        ),
        dict(
            x=1.0,
            y=1.02,
            xref="x",
            yref="paper",
            text="ERR = 1.0 (no excess)",
            showarrow=False,
            xanchor="center",
            yanchor="bottom",
            font=dict(family=FONT, size=9, color=FED_GREY_TEXT),
        ),
    ]

    fig.update_layout(
        **base_layout(
            2,
            "Excess readmission ratio by HRRP condition",
            (
                "All six tracked conditions exceed the 1.0 benchmark; differences are "
                "measured in thousandths of a ratio point."
            ),
            xaxis=dict(
                title="Average excess readmission ratio (ERR)",
                range=[0.9998, 1.0055],
                tickformat=".4f",
                gridcolor=FED_GRID,
                gridwidth=0.5,
                showline=True,
                linecolor=FED_DARK,
                linewidth=1,
            ),
            yaxis=dict(title="", showgrid=False),
            shapes=[
                dict(
                    type="line",
                    x0=1,
                    x1=1,
                    y0=0,
                    y1=1,
                    yref="paper",
                    line=dict(color=FED_DARK, width=1),
                )
            ],
            annotations=annotations,
            height=500,
        )
    )
    return fig


def chart3(own_df: pd.DataFrame) -> go.Figure:
    grouped = (
        own_df.groupby(["ownership_group", "penalty_tier"], as_index=False)["n"]
        .sum()
        .assign(
            pct=lambda d: d.groupby("ownership_group")["n"].transform(lambda s: s / s.sum())
        )
    )
    ownership_order = ["Non-Profit", "Government", "For-Profit"]
    tier_order = ["No Penalty", "Low", "Medium", "High"]

    fig = go.Figure()
    for tier in tier_order:
        part = grouped[grouped["penalty_tier"] == tier]
        fig.add_trace(
            go.Bar(
                name=tier,
                y=part["ownership_group"],
                x=part["pct"],
                orientation="h",
                marker_color=TIER_COLORS[tier],
                hovertemplate="<b>%{y}</b><br>"
                + tier
                + ": %{x:.1%}<extra></extra>",
            )
        )

    for_profit_high = grouped[
        (grouped["ownership_group"] == "For-Profit") & (grouped["penalty_tier"] == "High")
    ]["pct"]
    high_pct = float(for_profit_high.iloc[0]) if len(for_profit_high) else 0

    tier_label_x = {"No Penalty": 0.12, "Low": 0.38, "Medium": 0.62, "High": 0.86}
    tier_annotations = [
        dict(
            x=tier_label_x[tier],
            y=1.06,
            xref="paper",
            yref="paper",
            text=tier,
            showarrow=False,
            xanchor="center",
            font=dict(family=FONT, size=9, color=TIER_COLORS[tier]),
        )
        for tier in tier_order
    ]

    annotations = [
        dict(
            x=0.98,
            y="For-Profit",
            xref="paper",
            yref="y",
            text=f"High-tier share: {high_pct:.1%}",
            showarrow=False,
            xanchor="right",
            font=dict(family=FONT, size=9, color=FED_RED),
        ),
        *tier_annotations,
    ]

    fig.update_layout(
        **base_layout(
            3,
            "HRRP penalty tier distribution by hospital ownership type",
            (
                "Stacked shares sum to 100% within each ownership group. Ownership joined "
                "from CMS Hospital General Information (xubh-q36u)."
            ),
            barmode="stack",
            xaxis=dict(
                title="Share of hospital-condition pairs (%)",
                tickformat=".0%",
                range=[0, 1],
                gridcolor=FED_GRID,
                gridwidth=0.5,
                showline=True,
                linecolor=FED_DARK,
                linewidth=1,
            ),
            yaxis=dict(
                title="",
                categoryorder="array",
                categoryarray=ownership_order,
                showgrid=False,
            ),
            showlegend=False,
            annotations=annotations,
            height=460,
        )
    )
    return fig


def main() -> None:
    data_dir = REPO / "data"
    if not data_dir.exists():
        data_dir = ROOT / "public/data/articles/readmitted/data"

    state_df = pd.read_csv(data_dir / "hrrp_state_summary.csv")
    own_df = pd.read_csv(data_dir / "hrrp_ownership_condition.csv")

    save_chart(chart1(state_df), "chart1_states_penalized")
    save_chart(chart2(), "chart2_err_by_condition")
    save_chart(chart3(own_df), "chart3_penalty_by_ownership")


if __name__ == "__main__":
    main()

File: scripts/render-readmitted-charts.R
#!/usr/bin/env Rscript
# Export READMITTED charts (PNG + Plotly JSON) from monorepo articles/readmitted/.

suppressPackageStartupMessages({
  library(tidyverse)
  library(scales)
  library(plotly)
  library(jsonlite)
})

script_dir <- tryCatch(
  dirname(normalizePath(sub(
    "^--file=", "",
    commandArgs(trailingOnly = FALSE)[grep("^--file=", commandArgs(trailingOnly = FALSE))][1]
  ))),
  error = function(e) getwd()
)
repo_root <- normalizePath(file.path(script_dir, ".."))

source(file.path(repo_root, "scripts/r/artometrics_theme.R"))
art_init_fonts(repo_root)

# Match site cream (override theme default)
ART_CREAM <<- "#F2F0EB"

article_dir <- file.path(repo_root, "articles/readmitted")
data_dir <- file.path(article_dir, "data")
charts_dir <- file.path(article_dir, "charts")
if (!dir.exists(data_dir)) {
  stop("Missing articles/readmitted/data — run from monorepo root after promoting the article home")
}
dir.create(charts_dir, recursive = TRUE, showWarnings = FALSE)

public_img <- file.path(repo_root, "public/images/content/articles/readmitted/charts")
public_data <- file.path(repo_root, "public/data/articles/readmitted")
public_charts <- file.path(public_data, "charts")
public_csv <- file.path(public_data, "data")
dir.create(public_img, recursive = TRUE, showWarnings = FALSE)
dir.create(public_charts, recursive = TRUE, showWarnings = FALSE)
dir.create(public_csv, recursive = TRUE, showWarnings = FALSE)

NAT_AVG_PCT <- 48.1
HRRP_SOURCE <- "Source: CMS Hospital Readmissions Reduction Program (HRRP), FY2025 supplemental data (Dataset 9n3s-kdb3)"
FONT <- "Helvetica"

write_plotly_json <- function(fig, path) {
  built <- plotly_build(fig)
  payload <- list(
    data = built$x$data,
    layout = built$x$layout,
    config = list(
      displayModeBar = FALSE,
      displaylogo = FALSE,
      responsive = TRUE
    )
  )
  write_json(payload, path, auto_unbox = TRUE, pretty = TRUE, null = "null", na = "null")
  message("Wrote ", path)
}

copy_to_public <- function(name) {
  file.copy(
    file.path(charts_dir, paste0(name, ".png")),
    file.path(public_img, paste0(name, ".png")),
    overwrite = TRUE
  )
  file.copy(
    file.path(charts_dir, paste0(name, ".plotly.json")),
    file.path(public_charts, paste0(name, ".plotly.json")),
    overwrite = TRUE
  )
}

# Sync working CSVs to public for downloadable FILES section
file.copy(
  list.files(data_dir, pattern = "\\.csv$", full.names = TRUE),
  public_csv,
  overwrite = TRUE
)

save_png <- function(plot_obj, name, height = 7) {
  path <- file.path(charts_dir, paste0(name, ".png"))
  ggsave(
    filename = path,
    plot = plot_obj,
    width = 12,
    height = height,
    dpi = 300,
    bg = ART_CREAM,
    device = ragg::agg_png
  )
  message("Wrote ", path)
  invisible(path)
}

# ── Chart 1: States above national average ───────────────────────────────────

state_pct <- read_csv(file.path(data_dir, "hrrp_state_summary.csv"), show_col_types = FALSE) %>%
  filter(above_nat_avg) %>%
  arrange(pct_penalized) %>%
  mutate(
    state = fct_inorder(state),
    bar_color = art_bar_colors(n(), highlight_last = TRUE)
  )

nj_row <- state_pct %>% filter(as.character(state) == "NJ")

p1 <- ggplot(state_pct, aes(x = pct_penalized, y = state)) +
  geom_col(aes(fill = bar_color), width = 0.65, show.legend = FALSE) +
  scale_fill_identity() +
  geom_vline(xintercept = NAT_AVG_PCT, linetype = "dashed", color = ART_SECONDARY, linewidth = 1) +
  geom_text(
    aes(label = paste0(round(pct_penalized, 1), "%")),
    hjust = -0.15,
    size = 2.9,
    color = ART_DARK,
    family = FONT
  ) +
  annotate(
    "text",
    x = NAT_AVG_PCT,
    y = length(levels(state_pct$state)) + 0.6,
    label = paste0("National avg (", NAT_AVG_PCT, "%)"),
    color = ART_SECONDARY,
    size = 2.8,
    family = FONT
  ) +
  annotate(
    "segment",
    x = nj_row$pct_penalized,
    xend = nj_row$pct_penalized + 4,
    y = nj_row$state,
    yend = as.numeric(nj_row$state) + 0.55,
    arrow = arrow(length = unit(0.12, "inches"), type = "closed"),
    color = ART_HIGHLIGHT,
    linewidth = 0.45
  ) +
  annotate(
    "text",
    x = nj_row$pct_penalized + 4.2,
    y = as.numeric(nj_row$state) + 0.55,
    label = "NJ leads at 65.4%",
    hjust = 0,
    color = ART_HIGHLIGHT,
    size = 2.8,
    fontface = "bold",
    family = FONT
  ) +
  scale_x_continuous(
    labels = function(x) paste0(x, "%"),
    expand = expansion(mult = c(0, 0.18))
  ) +
  labs(
    title = "Which <span style='color:#C0392B;'>States</span> Have the Most Penalized Hospitals?",
    subtitle = "Above-average states only — share of hospital-condition pairs with ERR > 1.0",
    x = "% of Hospital-Condition Pairs with ERR > 1.0",
    y = NULL
  ) +
  theme_artometrics() +
  theme(
    plot.background = element_rect(fill = ART_CREAM, color = NA),
    panel.background = element_rect(fill = ART_CREAM, color = NA),
    panel.grid.major.y = element_blank()
  )

p1 <- finish_art_chart(
  p1,
  takeaway = "Geography isn't destiny — wealthy states lead the penalty list alongside the rural South",
  source = HRRP_SOURCE
)
save_png(p1, "chart1_states_penalized", height = 6.5 + nrow(state_pct) * 0.18)

# Plotly twin for interactive site chart
fig1 <- plot_ly(
  state_pct,
  x = ~pct_penalized,
  y = ~state,
  type = "bar",
  orientation = "h",
  marker = list(color = state_pct$bar_color, line = list(width = 0)),
  text = ~paste0(round(pct_penalized, 1), "%"),
  textposition = "outside",
  hovertemplate = "<b>%{y}</b><br>%{x:.1f}% of pairs penalized<extra></extra>"
) %>%
  layout(
    title = list(
      text = paste0(
        "<b>Which <span style='color:#C0392B;'>States</span> Have the Most Penalized Hospitals?</b>",
        "<br><span style='font-size:12px;color:#6B6B6B'>Above-average states only — share of hospital-condition pairs with ERR > 1.0</span>"
      ),
      x = 0,
      xanchor = "left"
    ),
    paper_bgcolor = ART_CREAM,
    plot_bgcolor = ART_CREAM,
    font = list(family = FONT, color = ART_DARK, size = 12),
    xaxis = list(
      title = "% of Hospital-Condition Pairs with ERR > 1.0",
      gridcolor = ART_MUTED,
      zeroline = FALSE,
      range = c(0, max(state_pct$pct_penalized) * 1.14)
    ),
    yaxis = list(title = "", categoryorder = "array", categoryarray = as.character(state_pct$state)),
    shapes = list(list(
      type = "line",
      x0 = NAT_AVG_PCT, x1 = NAT_AVG_PCT,
      y0 = 0, y1 = 1, yref = "paper",
      line = list(color = ART_SECONDARY, dash = "dash", width = 1.5)
    )),
    annotations = list(
      list(
        x = NAT_AVG_PCT, y = 1.02, xref = "x", yref = "paper",
        text = paste0("National avg (", NAT_AVG_PCT, "%)"),
        showarrow = FALSE, xanchor = "center", yanchor = "bottom",
        font = list(size = 9, color = ART_SECONDARY)
      ),
      list(
        x = 0.99, y = 0.01, xref = "paper", yref = "paper",
        text = "Artometrics", showarrow = FALSE,
        xanchor = "right", yanchor = "bottom",
        font = list(size = 11, color = ART_MID, family = "Georgia, serif")
      )
    ),
    margin = list(l = 60, r = 40, t = 90, b = 60),
    showlegend = FALSE
  )
write_plotly_json(fig1, file.path(charts_dir, "chart1_states_penalized.plotly.json"))
copy_to_public("chart1_states_penalized")

# ── Chart 2: ERR by condition ────────────────────────────────────────────────

condition_err <- read_csv(file.path(data_dir, "hrrp_condition_err.csv"), show_col_types = FALSE) %>%
  arrange(avg_err) %>%
  mutate(condition = fct_inorder(condition))

hip_row <- condition_err %>% filter(as.character(condition) == "Hip/Knee")

p2 <- ggplot(condition_err, aes(x = avg_err, y = condition)) +
  geom_segment(aes(x = 1.0, xend = avg_err, yend = condition), color = ART_MUTED, linewidth = 0.9) +
  geom_vline(xintercept = 1.0, color = ART_DARK, linewidth = 0.6) +
  geom_point(color = ART_HIGHLIGHT, size = 4.2, stroke = 0.4) +
  geom_text(
    aes(label = sprintf("%.5f", avg_err)),
    hjust = -0.25,
    size = 3.1,
    color = ART_DARK,
    family = FONT
  ) +
  annotate(
    "text",
    x = 1.0,
    y = length(levels(condition_err$condition)) + 0.55,
    label = "ERR = 1.0 (no excess)",
    color = ART_MID,
    size = 2.8,
    family = FONT
  ) +
  annotate(
    "segment",
    x = hip_row$avg_err,
    xend = hip_row$avg_err + 0.00055,
    y = hip_row$condition,
    yend = as.numeric(hip_row$condition) + 0.45,
    arrow = arrow(length = unit(0.12, "inches"), type = "closed"),
    color = ART_HIGHLIGHT,
    linewidth = 0.45
  ) +
  annotate(
    "text",
    x = hip_row$avg_err + 0.00058,
    y = as.numeric(hip_row$condition) + 0.45,
    label = "Nearly 2× the excess\nof the next condition",
    hjust = 0,
    color = ART_HIGHLIGHT,
    size = 2.7,
    fontface = "bold",
    lineheight = 0.95,
    family = FONT
  ) +
  scale_x_continuous(
    labels = function(x) sprintf("%.4f", x),
    expand = expansion(mult = c(0.01, 0.12))
  ) +
  labs(
    title = "The <span style='color:#C0392B;'>Hip/Knee</span> Problem: ERR by Condition",
    subtitle = "All six HRRP conditions exceed 1.0 — but the spread is measured in thousandths",
    x = "Average Excess Readmission Ratio (ERR)",
    y = NULL
  ) +
  theme_artometrics() +
  theme(
    plot.background = element_rect(fill = ART_CREAM, color = NA),
    panel.background = element_rect(fill = ART_CREAM, color = NA),
    panel.grid.major.y = element_blank()
  )

p2 <- finish_art_chart(
  p2,
  takeaway = "Elective joint surgery — not heart attacks — carries the highest excess readmission ratio",
  source = HRRP_SOURCE
)
save_png(p2, "chart2_err_by_condition", height = 5.2)

fig2 <- plot_ly(
  condition_err,
  x = ~avg_err,
  y = ~condition,
  type = "scatter",
  mode = "markers",
  marker = list(color = ART_HIGHLIGHT, size = 12),
  text = ~sprintf("%.5f", avg_err),
  hovertemplate = "<b>%{y}</b><br>ERR %{x:.5f}<extra></extra>"
) %>%
  add_segments(
    x = 1.0, xend = ~avg_err,
    y = ~condition, yend = ~condition,
    line = list(color = ART_MUTED, width = 2),
    showlegend = FALSE,
    hoverinfo = "skip"
  ) %>%
  layout(
    title = list(
      text = paste0(
        "<b>The <span style='color:#C0392B;'>Hip/Knee</span> Problem: ERR by Condition</b>",
        "<br><span style='font-size:12px;color:#6B6B6B'>All six HRRP conditions exceed 1.0 — but the spread is measured in thousandths</span>"
      ),
      x = 0, xanchor = "left"
    ),
    paper_bgcolor = ART_CREAM,
    plot_bgcolor = ART_CREAM,
    font = list(family = FONT, color = ART_DARK, size = 12),
    xaxis = list(title = "Average Excess Readmission Ratio (ERR)", gridcolor = ART_MUTED),
    yaxis = list(title = "", categoryorder = "array", categoryarray = as.character(condition_err$condition)),
    shapes = list(list(
      type = "line", x0 = 1, x1 = 1, y0 = 0, y1 = 1, yref = "paper",
      line = list(color = ART_DARK, width = 1)
    )),
    annotations = list(list(
      x = 0.99, y = 0.01, xref = "paper", yref = "paper",
      text = "Artometrics", showarrow = FALSE,
      xanchor = "right", yanchor = "bottom",
      font = list(size = 11, color = ART_MID, family = "Georgia, serif")
    )),
    margin = list(l = 100, r = 40, t = 90, b = 60),
    showlegend = FALSE
  )
write_plotly_json(fig2, file.path(charts_dir, "chart2_err_by_condition.plotly.json"))
copy_to_public("chart2_err_by_condition")

# ── Chart 3: Penalty tier by ownership ─────────────────────────────────────────

plot_data <- read_csv(file.path(data_dir, "hrrp_ownership_condition.csv"), show_col_types = FALSE) %>%
  group_by(ownership_group, penalty_tier) %>%
  summarise(n = sum(n), .groups = "drop") %>%
  group_by(ownership_group) %>%
  mutate(pct = n / sum(n)) %>%
  ungroup() %>%
  mutate(
    penalty_tier = factor(penalty_tier, levels = c("No Penalty", "Low", "Medium", "High")),
    ownership_group = factor(ownership_group, levels = c("Non-Profit", "Government", "For-Profit"))
  )

for_profit_high <- plot_data %>%
  filter(ownership_group == "For-Profit", penalty_tier == "High") %>%
  pull(pct)
high_pct <- if (length(for_profit_high)) for_profit_high[[1]] else 0

p3 <- ggplot(plot_data, aes(x = pct, y = ownership_group, fill = penalty_tier)) +
  geom_col(position = "fill", width = 0.62) +
  scale_x_continuous(labels = percent_format(accuracy = 1)) +
  scale_fill_manual(values = ART_TIER_COLORS, name = "Penalty Tier") +
  annotate(
    "text",
    x = 0.98,
    y = "For-Profit",
    label = paste0("High-tier share: ", percent(high_pct, accuracy = 0.1)),
    hjust = 1,
    color = ART_HIGHLIGHT,
    size = 2.9,
    fontface = "bold",
    family = FONT
  ) +
  labs(
    title = "<span style='color:#C0392B;'>For-Profit</span> Hospitals Carry More Penalty Weight",
    subtitle = "Penalty tier distribution by ownership — HCA, Tenet, and Steward sit in the for-profit column",
    x = "Share of Hospital-Condition Pairs",
    y = NULL
  ) +
  theme_artometrics() +
  theme(
    plot.background = element_rect(fill = ART_CREAM, color = NA),
    panel.background = element_rect(fill = ART_CREAM, color = NA),
    panel.grid.major.y = element_blank(),
    legend.position = "top",
    legend.justification = "right",
    legend.box.just = "right",
    legend.direction = "horizontal"
  )

p3 <- finish_art_chart(
  p3,
  takeaway = "For-profits carry more penalty weight, but every ownership type still has a majority in the no-penalty band",
  source = paste0(HRRP_SOURCE, " · joined to CMS Hospital General Information (xubh-q36u) for ownership")
)
save_png(p3, "chart3_penalty_by_ownership", height = 4.8)

# Stacked horizontal bars for Plotly
tier_levels <- c("No Penalty", "Low", "Medium", "High")
own_levels <- c("Non-Profit", "Government", "For-Profit")
fig3 <- plot_ly()
for (tier in tier_levels) {
  slice <- plot_data %>% filter(penalty_tier == tier)
  # align to ownership order
  vals <- sapply(own_levels, function(o) {
    hit <- slice$pct[slice$ownership_group == o]
    if (length(hit)) hit[[1]] else 0
  })
  fig3 <- fig3 %>% add_trace(
    type = "bar",
    orientation = "h",
    name = tier,
    x = vals,
    y = own_levels,
    marker = list(color = unname(ART_TIER_COLORS[tier])),
    hovertemplate = paste0("<b>%{y}</b><br>", tier, ": %{x:.1%}<extra></extra>")
  )
}
fig3 <- fig3 %>% layout(
  barmode = "stack",
  title = list(
    text = paste0(
      "<b><span style='color:#C0392B;'>For-Profit</span> Hospitals Carry More Penalty Weight</b>",
      "<br><span style='font-size:12px;color:#6B6B6B'>Penalty tier distribution by ownership</span>"
    ),
    x = 0, xanchor = "left"
  ),
  paper_bgcolor = ART_CREAM,
  plot_bgcolor = ART_CREAM,
  font = list(family = FONT, color = ART_DARK, size = 12),
  xaxis = list(title = "Share of Hospital-Condition Pairs", tickformat = ".0%", gridcolor = ART_MUTED),
  yaxis = list(title = "", categoryorder = "array", categoryarray = own_levels),
  legend = list(orientation = "h", y = 1.12, x = 1, xanchor = "right"),
  annotations = list(list(
    x = 0.99, y = 0.01, xref = "paper", yref = "paper",
    text = "Artometrics", showarrow = FALSE,
    xanchor = "right", yanchor = "bottom",
    font = list(size = 11, color = ART_MID, family = "Georgia, serif")
  )),
  margin = list(l = 100, r = 40, t = 100, b = 60)
)
write_plotly_json(fig3, file.path(charts_dir, "chart3_penalty_by_ownership.plotly.json"))
copy_to_public("chart3_penalty_by_ownership")

cat("All readmitted R charts exported to", charts_dir, "and synced to public/\n")
